yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/Matchers.java

```diff
@@ -471,12 +477,12 @@ private static ImmutablePair<ResultSetMatchResult, ResultSetPrettyPrinter> match
             }
             if (!found) {
                 printCurrentAndRemaining(actual, resultSetPrettyPrinter);
-                return ImmutablePair.of(ResultSetMatchResult.fail(String.format("result row at %d does not match any expected records", i)), resultSetPrettyPrinter);
+                return ImmutablePair.of(ResultSetMatchResult.fail(String.format("actual row at %d does not match any expected records", actualRowsCounter)), resultSetPrettyPrinter);
             }
-            i++;
+            actualRowsCounter++;
         }
         if (!expectedAsMultiSet.isEmpty()) {
-            return ImmutablePair.of(ResultSetMatchResult.fail("result does not contain all expected rows"), null);
+            return ImmutablePair.of(ResultSetMatchResult.fail(String.format("result does not contain all expected rows, expected %d row(s), got %d row(s) instead.", expectedRowCount, actualRowsCounter - 1)), null);
```

Comment:
I understand that this is tests, but I think we want to refrain from `String.format` (especially without a locale) if we can avoid it. That does seem to be a larger problem in this file, though

Reply:
I made all the `String.format` calls here use `Locale.ROOT`. I will propose a separate PR that extends this fix to the entire codebase.